### ISSUE TITLE
combining dynamically and statically named chunks behaves weirdly

### ISSUE THREAD

**Issue by aem** [webpack/webpack#8026]
I'm not sure if this is a feature request because this currently WAD or a bug report because it's broken, so here goes nothing.

tl;dr: combining dynamically named chunks (e.g. `import( /* webpackChunkName: "chunk-[request]" */ './path/to/' + chunkname)`) and a statically named chunk of the same name results in two chunks instead of one. i've put together a simple repro here: https://github.com/aem/chunk-name-bug-repro

The same way you can use dynamic import to add multiple codepaths to the same chunk by giving them the same chunk name, I'd assume that if you have a dynamically-generated chunk name but know what the name of that chunk will eventually be that you should still be able to combine those chunks into one larger chunk without emitting an additional asset.

this is reproducible in both `webpack@3` and `webpack@latest`

simple repro, pulled out of the repo that I linked above:

```js
// index.js
// assumes a `routes` directory containing a.js b.js and c.js exists
function importRoute(routeName) {
  return import(/* webpackChunkName: "chunk-[request]" */ `../routes/${routeName}`).then(
    module => module.default.render()
  );
}

importRoute('a');
import(/* webpackChunkName: "chunk-a" */ '../routes/b').then(module =>
  module.default.render())
```

### expected chunks

main.js
chunk-a.js (containing route a and route b)
chunk-b.js
chunk-c.js

### actual chunks

main.js
chunk-a.js (containing route a and route b)
chunk-a-js.js (containing route a)
chunk-b.js
chunk-c.js

**Comment by sokra** [comment]
The context matches these requests:

* `./a`
* `./a.js`
* `./b`
* `./b.js`
* `./c`
* `./c.js`

This results in these chunks:

* `chunk-a`: `a.js`
* `chunk-a-js`: `a.js`
* `chunk-b`: `b.js`
* `chunk-b-js`: `b.js`
* `chunk-c`: `c.js`
* `chunk-c-js`: `c.js`

You additional imports adds `b.js` to `chunk-a`

Now chunks with the same contents are merged (preferring the shorter name)

* `chunk-a`: `a.js` + `b.js`
* `chunk-a-js`: `a.js`
* `chunk-b`: `b.js`
* `chunk-c`: `c.js`

That's why it behaves this way.

---

A fix could be to create a single chunk name per unique file.

**Comment by aem** [comment]
I didn't realize that chunks were generated with and without the `-js` extension and then merged during optimization. Makes perfect sense that's it's working this way. I'd obviously opt to create the single chunk per unique file, as the double-chunk generation is messing a bit with our commons chunk, but I'm not sure if there are users/plugins that depend on multiple chunks being generated.

**Comment by alexander-akait** [comment]
Closing due to inactivity. Please test with latest version and feel free to reopen if still regressions. Thanks!
